File: java/src/DirectionSouth.java

public class DirectionSouth implements IDirection
{
	public IDirection left()
	{
		return new DirectionEast();
	}
	
	public IDirection right()
	{
		return new DirectionWest();
	}
	
	public Position move(Position position)
	{
		return position;
	}

	public String name()
	{
		return "SOUTH";
	}
}

File: ruby/src/DirectionSouth.rb
require_relative "IDirection.rb"
require_relative "DirectionWest.rb"
require_relative "DirectionEast.rb"

class DirectionSouth < IDirection
  
  def name()
    return "SOUTH"
  end
  
  def left()
    return DirectionEast.new()
  end
  
  def right()
    return DirectionWest.new()
  end
  
  def move(position)
    position.y_position = position.y_position - 1
    return position
  end
end
